activemq-broker/src/main/java/org/apache/activemq/broker/jmx/BrokerView.java

```diff
@@ -280,11 +280,21 @@ public ObjectName[] getTopics() {
         return safeGetBroker().getTopicsNonSuppressed();
     }
 
+    @Override
+    public int getTotalTopicsCount() {
+        return getTopics().length;
+    }
+
     @Override
     public ObjectName[] getQueues() {
         return safeGetBroker().getQueuesNonSuppressed();
     }
 
+    @Override
+    public int getTotalQueuesCount() {
+        return getQueues().length;
```

Comment:
This still calls a method that creates a new array of objects. Why not just return the value of the size from the map?

Reply:
Good callout, I believe the new array of objects you are referring to is the one created in `onlyNonSuppressed`  which copy over the non suppressed queues or topics already maintained in the `ManagedRegionBroker`.  Are you saying I should just simply return the size of `private final Map<ObjectName, DestinationView> queues = new ConcurrentHashMap<>();` in `ManagedRegionBroker`?

My concern is that it will not agree with the queue or topic list in the `queues` and `topics` MBean attribute, and those are non suppressed which should be the correct list.

The optimization done at this PR is mainly for user application. Because they will need to read that ObjectName array into their memory space even all they need is the count of `queues` or `topics`. So if this PR is merged as it is, the user application memory space will not be bloated (even tho as your pointed out, on the broker side, it will still need to create an array of those objects on the broker process memory space).

Reply:
I'm saying the suppression info shouldn't be considered at all. If there are 128 queues, and 28 suppressed, the user is still going to want to know there are 128 queues and not 100.

Reply:
I disagree @mattrpav, I think this method should be consistent with the `getQueues()` that uses `getQueuesNonSuppressed()`.

So, I think what @kenliao94 does makes sense.

I'm +1 with this change.

Reply:
@jbonofre none of the other "Total*" metrics take in account the suppression, why would we start with this one?

Reply:
The current implementation is a -1 for me. There is _zero_ reason to allocate an array, and then shove a bunch of objects into it just to get a count.

Proposed path forward:

The total queue count should simply come from the queue region and do a quick size on an existing collection.

Reply:
What about:

```
    public ObjectName[] getQueues() {
        return safeGetBroker().getQueuesNonSuppressed();
    }
```

?

Reply:
That creates an array of ObjectName[] unnecessarily. 

This is a zero allocation operation:

```
managedBroker.getQueueRegion().getDestinationMap().size()
```

Reply:
I got your point, but it looks more "logic" to use the existing method to me. The `ObjectName[]` is not very costly.

Reply:
Just to explain my thought: currently, an user who wants to get the total number of queues gonna use `getQueues()` and count himself. So, the rationale of this change is to provide a convenient to directly provide the queues count.

It seems "logic" to me to use the same method as the user would use to do the count.

Reply:
How is this logic? This is simply walking pointer references to get a size off a collection.

```
managedBroker.getQueueRegion().getDestinationMap().size()
```

Reply:
The logic is the consistency between `getQueues()` and `getQueuesCount()` operations. An user might be confused if he counts with `getQueues()` and compare with `getQueuesCount()`: if he has different numbers, he would be confused.

Reply:
Right! That is exactly my point. The getQueues pulls in what is _registered_ in JMX, not what is present in the broker. JMX suppression is to prevent overhead caused by registering objects into JMX -- specifically for environments with _lots_ of queues and queues that are one-time-use queues.

If a user enable suppression for performance reasons, they still need base metrics. The BrokerView metrics should be consistent and provide -- what is the in broker?

I think the confusion is around what does the getQueues() method mean? getQueues() _not_ "What queues are in the broker?" -- getQueues() is "what queues are _registered in JMX_?"

Reply:
That's also my point: as `getQueues()` MBean operation returns the JMX registered queues, the `getQueuesCount()` should also return the JMX registered queues count.

Reply:
@jbonofre I disagree. The behavior you are describing would be inconsistent with how other metrics at the broker-level operate. Specifically, the connection metrics.

Reply:
I agree it's "different" compare to the behavior of other MBeans (like connection), BUT it's consistent inside this specific MBean.

Reply:
getQueues() <-- returns ObjectName[], an array of JMX ObjectNames for queues that are registered, this isn't _all queues in the broker_. If anything, this method is mislabeled.

Use cases:

Use Case 1: 128 queues, no jmx mbean suppression
Use Case 2: 128 queues, 128 queues suppressed
Use Case 3: 128 queues, 28 queues suppressed

Current proposal:

| Use Case | # queues | # suppressed | getQueues() result | getTotalQueueCount() |
| ------------- | ------------- | ------------- | ------------- | ------------- |
| 1 | 128 | 0 | 128 ObjectName[] | 128 |
| 2 | 128 | 128 | 0 ObjectName[] | 0 |
| 3 | 128 | 28 | 28 ObjectName[] | 28 |

Reply:
I'm sorry, but using my user hat, I still think the current proposal makes more sense to me 😄 

@cshannon as we need an arbiter here, what do you think ? 
@kenliao94 what's your thoughts about that ?

Reply:
Maybe it's more a matter of operation name. Both operations could make sense (depending of the user cases).

Reply:
> Maybe it's more a matter of operation name. Both operations could make sense (depending of the user cases).

I'd be fine with adding the metric, while not allocating the array to do it ;-)

getRegisteredQueueCount()  <-- returns ObjectName[] length

```
public int getRegisteredQueueCount() {
   return managedBroker.getQueueViews().size()
```

Reply:
It looks like a good consensus to me 😄 

@kenliao94 thoughts ?

Reply:
Can I be greedy and request the getTotalQueueCount() also be added?

Reply:
Thanks for the input @mattrpav and @jbonofre  I see the points of having different names to avoid confusion. I will rename, avoid allocating an array and add the new count metrics.

Reply:
Done.